src/Http/Http.Abstractions/src/PathString.cs

```diff
@@ -172,8 +173,10 @@ private static string ToEscapedUriComponent(string value, int i)
         /// <returns>The resulting PathString</returns>
         public static PathString FromUriComponent(string uriComponent)
         {
-            // REVIEW: what is the exactly correct thing to do?
-            return new PathString(Uri.UnescapeDataString(uriComponent));
```

Comment:
uriComponent (utf16) -> pathBuffer (utf8) -> PathString (utf16)

Can we avoid one of these by writing a UrlDecoder method for `Span<char>`?

Reply:
That makes sense, I'll take a look.

Reply:
You're right that the UTF-8 step should be unnecessary, the input _should_ be escaped ascii so Span<char> should work.

Reply:
Just to confirm, this way we say that the utf-16 `string uriComponent` input is just a representation of an UTF8+UrlEncoded string, while the output is a PathString (holding a truly UTF-16 string)?

When I add an overload for `Span<char>` in UrlDecoder in my local change, I end up having quite some duplication, I cannot get rid of in a polite manner. I'll push an update soon.

Reply:
> Just to confirm, this way we say that the utf-16 `string uriComponent` input is just a representation of an UTF8+UrlEncoded string, while the output is a PathString (holding a truly UTF-16 string)?

Correct.

> When I add an overload for `Span<char>` in UrlDecoder in my local change, I end up having quite some duplication, I cannot get rid of in a polite manner. I'll push an update soon - if that direction makes sense, I will clean it up.

The duplication is unfortunate, but it does accomplish what we want. We might be able to avoid duplicating the tests by having the same test cover both versions of the API. That would also have the benefit of keeping them in sync.

Reply:
Tests added for UrlDecoder